### DISCUSSION TITLE
What I Tell Every Project About Linux

### DISCUSSION THREAD

**Discussion by richardstevenhack** [Stability-AI/StableSwarmUI#407] [Ideas]
1) If you want to run a python project on Linux, use pipx because some Linux distros, like openSUSE Tumbleweed are MANAGED python environments and do not let you just randomly install python libraries and the like. Pipx automates setting up a local user python environment and ensuring that there is a path to that project so the user can find it.

2) If you want to run ANYTHING on Linux that is NOT DIRECTLY SUPPORTED by ALL Linux distributions, create a Flatpak, an AppImage or a Snap (in that order of preference). Otherwise a rolling release distribution will crash your program sooner or later. In a pinch, run it in Docker, but because Docker Desktop is not available everywhere and Docker command line is a kluge, that is not a great option compared to all-in-one deployments.

For me, these conditions for using any Python project are non-negotiable.

**Comment by mcmonkey4eva** [comment]
Swarm is a C# project, the comfy backend is the python bit, so direct most commentary about python management to https://github.com/comfyanonymous/ComfyUI

Swarm's installer for comfy puts it in a venv and exclusively installs to the venv, not to global/system python installs

Swarm does include docker https://github.com/Stability-AI/StableSwarmUI?tab=readme-ov-file#running-with-docker
